### DISCUSSION TITLE
Testing Custom Webterm Commands

### DISCUSSION THREAD

**Discussion by matthewauld** [beakerbrowser/beaker#1707] [General] [answered]
So one problem I haven't been able to figure out is how to test webterm commands. Right now, to see if a change in my code has worked, i have to run `commands uninstall $@` , `commands install $@` and then `myCommand`. There doesn't seem to be a way to automate this and have it execute in the webterm context- unless I am missing something.

I am going to try to put together a "webterm simulator" class that simulates the "this" object, so you can just import your commands in JS and run them in a script, but I was wondering if anyone had a better way I had overlooked?

**Comment by pfrazee** [comment, marked as answer]
Yeah one other thing you can do right now is close and re-open webterm.

It would not be hard to add a command which reloads the environment. Maybe "commands reload" ?

**Reply by pfrazee** [reply]
Opened an issue: https://github.com/beakerbrowser/beaker/issues/1708

**Reply by matthewauld** [reply]
That is great! super helpful. I was thinking more about running test cases automatically, but I think I can do it by writing the test cases in JS (importing the command files) and running it from a webterm command using `eval` so it is executed in a webterm environment. I'll keep playing around with it :)
